pkg/pillar/types/assignableadapters.go

```diff
@@ -455,3 +464,61 @@ func (aa *AssignableAdapters) ExpandControllers(log *base.LogObject, list []*IoB
 	}
 	return elist
 }
+
+func getDeviceNameFromPciID(pciID string) (string, error) {
+	// e.g., ls /sys/bus/pci/devices/<pciID>/nvme/
+	//  -> nvme0
+	deviceName := ""
+	nvmePath, err := ioutil.ReadDir(sysfsPciDevices + pciID + "/nvme")
+	if err != nil {
+		return "", err
+	}
+	for _, file := range nvmePath {
+		deviceName = file.Name()
+	}
+	return deviceName, nil
+}
+
+// NVMEIsUsed checks if an NVME device is in a ZFS pool or mounted
```

Comment:
I think the comment should also say that during boot if we don't yet know, it will return true.
And maybe the name should reflect that by calling it NVMEIsMaybeUsed()?